installer/scripts/service_control

```diff
@@ -69,49 +176,107 @@ verify_privileges()
     fi
 }
 
-is_omsagent_running()
+this_omsagent_running()
 {
-    verify_privileges
+    if [ -f $PIDFILE ]; then
+        ps -p `cat $PIDFILE` | grep -q omsagent
+        WS_STATUS=$?
+    else
+        WS_STATUS=$FILE_NOT_FOUND
+    fi
+    return $WS_STATUS
+}
 
+is_omsagent_running()
+{
     # Returns 1 if 'omsagent' is running, 0 otherwise
-    [ -f $PIDFILE ] || return 0
-    ps -p `cat $PIDFILE` | grep -q omsagent
-    STATUS=$?
+    #### Keeping return value on this as it originally was.  I deduce
+    #### that because this value is returned with the is-running qualifier
+    #### to service_control, that there are scripts in other languages
+    #### that use the more conventional values for booleans.  Meanwhile,
+    #### I recommend deprecation in all but the external interface.  XC
+    if this_omsagent_running; then
+        return 1
+    else
+        return 0
+    fi
+} 
+
+remove_PIDfile_unless_omsagent_running()
+{
 
-    # Kill PID file if we're not running any longer
-    if [ $STATUS -ne 0 ]; then
-        rm -f $PIDFILE
+    WS_STATUS=0
+    if this_omsagent_running; then
         return 0
-    else
-        return 1
     fi
+    rm -f $PIDFILE # NOTE:  definition of PIDFILE is dependent on WS.
+    if [ $? -ne 0 ]; then
+        WS_STATUS=2
+        return $WS_STATUS
+    fi
+    return 1
 }
 
 exit_if_agent_not_onboarded()
 {
     if [ ! -f $OMSADMIN_CONF ]; then
-        echo "Warning: Agent is not onboarded. omsagent cannot be registered as a service."
-        exit 0 
+        echo "FATAL: Agent is not onboarded. omsagent cannot be registered as a service."
+        WS_STATUS=$FILE_NOT_FOUND
+        exit $WS_STATUS
     fi
 }
 
+wait_until_omsagent_is_running()
+{
+    # Required parameter: Number of seconds to wait for agent to become running.
+    if [ -z "$1" -o "$1" -le 0 ]; then
+        echo "Function \"wait_until_omsagent_stops\" called with invalid parameter"
+        WS_STATUS=1
+        exit $WS_STATUS
+    fi
+
+    COUNTER=$(( $1 * 2 )) # Since we sleep 0.5 seconds, compute number of seconds
+    while [ $COUNTER -gt 0 ]; do
+        if ! this_omsagent_running; then
+            COUNTER=$(( $COUNTER - 1 ))
```

Comment:
Do you have a preference between this method of arithmetic and using `expr`? I know a certain amount of our arithmetic code that has been in use for a while uses the former method (see [this example](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/scripts/omsadmin.sh#L312)). I would suggest using only only method, and I personally prefer the `$(())` method.

Reply:
No, I like to see consistent usage.  I saw that in one location, so I used it here.  I realize now we have both.  I suggest, again, that we regularize such things when we divide the library from the script.

Reply:
I think what is most omnipresent is probably the most important, and after that what is clearer or more acceptable to the group, as the project group will mostly read this, despite its being public.  I have been trying to pull stuff up I've seen elsewhere in the code, so I probably saw both somewhere.